### PR TITLE
Merge VS Code 1.31.1

### LINKED ISSUE #4522: TS Strict Null Checks complains about missing file
See error below - can we update this file? On next merge will this happen again?
```
[16:22:29] Starting compilation in watch mode...

error TS6053: File '/Users/projects/azuredatastudio/src/vs/workbench/parts/terminal/common/terminalMenu.ts' not found.

[16:22:50] Found 1 error. Watching for file changes.
```

### PR DESCRIPTION
**Pull request by MattIrv**
This PR merges VS Code 1.31.1. 

Unfortunately I committed in the wrong order so all my build/regression fixes show up in this PR in the same commit as the merge itself but if you want to do any reviewing spot checks are appreciated especially on things marked with SQL CARBON EDIT

Fixes #4522

### CODE REVIEW

**Comment by anthonydresser** [comment]
When I was messing around with this I was having typings conflictions between node v10 and the html parser stuff we have. Did you find a solution to that?

**Comment by MattIrv** [comment]
Yes the htmlparser type problem was fixed by setting the "@types/node" resolution in `package.json` to "10.12.12". I'll double check that I can build successfully using node 10 though since I've mostly been using 8.

File: extensions/html-language-features/client/src/customData.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (anthonydresser) [inline extensions/html-language-features/client/src/customData.ts]: Should this be deleted?

File: extensions/typescript-language-features/src/test/cachedResponse.test.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
```
Comment (anthonydresser) [inline extensions/typescript-language-features/src/test/cachedResponse.test.ts]: Should these be deleted?

**Comment by MattIrv** [comment]
Fixed everything that was pointed out and got the builds succeeding. Thanks for reviewing Anthony

File: resources/completions/bash/code [hunk 3]
```diff
@@ -0,0 +1,61 @@
+_code()
+{
```
Comment (anthonydresser) [inline resources/completions/bash/code]: We are probably going to want to update this for our purposes

Comment (anthonydresser) [reply resources/completions/bash/code]: At some point; maybe make a engineering task for this

File: resources/completions/zsh/_code [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#compdef code
+
+local arguments
+
+arguments=(
```
Comment (anthonydresser) [inline resources/completions/zsh/_code]: same with this completion file.

**Comment by kburtram** [comment]
@MattIrv let's hold off on merging this until after we create the release/1.5 branch later this week.

**Comment by MattIrv** [comment]
This PR is ready to go whenever that branch is created. It will probably have some more merge conflicts once more changes come in to master

**Comment by anthonydresser** [comment]
Could you put a short comment in the original comment about what the major changes that impact us are? It would help identify where to look in the PR.

**Pull request merged**